### PR TITLE
simplify MyAvatar::deriveBodyFromHMDSensor() take 2

### PR DESCRIPTION
**Pull request by AndrewMeadows**
This is another attempt to salvage some code I've been working on for a while.

We are now using a more correct measure of "eyes to hips" using the current animation state rather than an approximation.

I also simplified the logic that decides when to start/stop "straightening" the animation state, which happens when the location of the hips deviates significantly from that of the avatar's position.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6733767d8b41f16f553ab546aa782f4b2e695674 available at: https://gist.github.com/9d91e9ee6181bd8b6a01<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6391.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6391.dmg

File: libraries/animation/src/Rig.cpp [hunk 1]
```diff
@@ -407,6 +407,24 @@ void Rig::calcAnimAlpha(float speed, const std::vector<float>& referenceSpeeds,
...
+        int hipsIndex = _animSkeleton->nameToJointIndex(QString("Hips"));
+        int headIndex = _animSkeleton->nameToJointIndex(QString("Head"));
+        if (hipsIndex >= 0 && headIndex > 0) {
+            glm::vec3 rightEye = _animSkeleton->getAbsolutePose(rightEyeIndex, poses).trans;
+            glm::vec3 leftEye = _animSkeleton->getAbsolutePose(leftEyeIndex, poses).trans;
+            glm::vec3 hips = _animSkeleton->getAbsolutePose(hipsIndex, poses).trans;
+            _eyesInRootFrame = 0.5f * (rightEye + leftEye) - hips;
+        }
+    }
+}
```
Comment (AndrewMeadows) [inline libraries/animation/src/Rig.cpp]: We compute and cache the position of the eyes after the all of the animation blending and IK is done.  This value is consumed by MyAvatar when it computes the "bodyToSensor" transform.

File: interface/src/avatar/MyAvatar.cpp [hunk 2]
```diff
@@ -1994,53 +1975,19 @@ glm::quat MyAvatar::getWorldBodyOrientation() const {
...
-    // eyeToNeck offset is relative full HMD orientation.
-    // while neckToRoot offset is only relative to HMDs yaw.
-    glm::vec3 eyeToNeck = hmdOrientation * (localNeck - localEyes);
-    glm::vec3 neckToRoot = hmdOrientationYawOnly * -localNeck;
-    glm::vec3 bodyPos = hmdPosition + eyeToNeck + neckToRoot;
-
-    // avatar facing is determined solely by hmd orientation.
-    return createMatFromQuatAndPos(hmdOrientationYawOnly, bodyPos);
+    return glm::mat4();
 }
```
Comment (AndrewMeadows) [inline interface/src/avatar/MyAvatar.cpp]: Here we use a more correct "eyes to hips" value to compute the "avatar center in current HMD frame".

File: interface/src/avatar/MyAvatar.cpp [hunk 3]
```diff
@@ -177,9 +177,8 @@ void MyAvatar::reset(bool andReload) {
 
     // Reset dynamic state.
     _wasPushing = _isPushing = _isBraking = _billboardValid = false;
-    _isFollowingHMD = false;
-    _hmdFollowVelocity = Vectors::ZERO;
-    _hmdFollowSpeed = 0.0f;
+    _followVelocity = Vectors::ZERO;
+    _followSpeed = 0.0f;
```
Comment (AndrewMeadows) [inline interface/src/avatar/MyAvatar.cpp]: **_isFollowingHMD** is gone.  The avatar is "following" its hips whenever **_followSpeed** is not zero.

**Comment by AndrewMeadows** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6733767d8b41f16f553ab546aa782f4b2e695674 available at: https://gist.github.com/9d91e9ee6181bd8b6a01<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6391.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6391.dmg

**Comment by howard-stearns** [comment]
Problem (as discussed) with a smooth stutter as you walk in hmd while looking at your feet.
Not present in master.

**Comment by howard-stearns** [comment]
I realize they didn't originate with this PR, but when you get a chance (in some future pr), could we document what followSpeed and followVelocity are (and why they're different)?

**Pull request merged**